File: frontend/ultima-ng-19.0.0/src/app/interface/localisation.ts
export interface ILocalisation {
    localisationId: number;
    localisationUuid: string;
    adresseComplete: string;
    latitude: number | null;
    longitude: number | null;
    description: string | null;
    quartierId: number | null;
    quartierUuid: string | null;
    quartierLibelle: string | null;
    communeUuid: string | null;
    communeLibelle: string | null;
    villeUuid: string | null;
    villeLibelle: string | null;
    regionUuid: string | null;
    regionLibelle: string | null;
    createdAt: Date | string;
    updatedAt: Date | string;
}

export interface ILocalisationCreateRequest {
    quartierUuid?: string | null;
    adresseComplete: string;
    latitude?: number | null;
    longitude?: number | null;
    description?: string | null;
}

export interface ILocalisationUpdateRequest {
    quartierUuid?: string | null;
    removeQuartier?: boolean;
    adresseComplete: string;
    latitude?: number | null;
    longitude?: number | null;
    description?: string | null;
}

// Pour OpenStreetMap/Nominatim Autocomplete
export interface IPlacePrediction {
    placeId: string;
    description: string;
    latitude?: number;
    longitude?: number;
    type?: string;
    category?: string;
    // Champs spécifiques Guinée
    quartier?: string;
    commune?: string;
    ville?: string;
    region?: string;
}

export interface IPlaceDetails {
    address: string;
    latitude: number;
    longitude: number;
}

File: microservers/billetterieservice/src/main/java/io/multi/billetterieservice/domain/Localisation.java
package io.multi.billetterieservice.domain;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Localisation {

    private Long localisationId;
    private String localisationUuid;
    private Long quartierId;
    private String adresseComplete;
    private BigDecimal latitude;
    private BigDecimal longitude;
    private String description;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;

    // Champs additionnels pour les jointures (nullable car quartier_id est nullable)
    private String quartierUuid;
    private String quartierLibelle;
    private String communeUuid;
    private String communeLibelle;
    private String villeUuid;
    private String villeLibelle;
    private String regionUuid;
    private String regionLibelle;
}

